src/wireviz/Harness.py

```diff
@@ -105,11 +105,15 @@ def create_graph(self) -> Graph:
 
             if connector.style != 'simple':
                 pinlist = []
-                for pin, pinlabel in zip(connector.pins, connector.pinlabels):
+                for counter, pin, pinlabel in zip(range(0, len(connector.pins)), connector.pins, connector.pinlabels):
                     if connector.hide_disconnected_pins and not connector.visible_pins.get(pin, False):
                         continue
                     pinlist.append([f'<td port="p{pin}l">{pin}</td>' if connector.ports_left else None,
-                                    f'<td>{pinlabel}</td>' if pinlabel else '',
+                                    f'<td>{pinlabel}</td>' if pinlabel else None,
+                                    # f'<td>{counter}</td>',  # for debugging only
+                                    f'<td sides="tbl">{connector.pincolors[counter].replace("__","")}</td>' if connector.pincolors and connector.pincolors[counter] else None,
+                                    # f'<td bgcolor="{wv_colors.translate_color(connector.pincolors[counter], "HEX")}" width="4"></td>' if connector.pincolors else None,
+                                    f'<td sides="tbr">{html_color_icon(connector.pincolors[counter])}</td>' if connector.pincolors and connector.pincolors[counter] else None,
```

Comment:
> * Is it possible to reduce the space between color code and bullet? They take a lot of space together.

I tried, but so far, without any real luck. I tried specifying the `width` of the cell, but it was not respected. Using `fixedsize="true"` also produced weird results (smaller cell, but huge gap _between_ cells).

It's about the auto-scaling of the different nested tables; the same happens for the color mark for the entire connector under certain circumstances. See the following dummy render. Everything is scaled proportionally to fit the largest element in the table, distorting smaller elements.

![test](https://user-images.githubusercontent.com/2447251/89005978-ff50bd00-d305-11ea-98bb-d351c7056d46.png)

---

I'll address the other points in upcoming commits.

Reply:
> It's about the auto-scaling of the different nested tables; the same happens for the color mark for the entire connector under certain circumstances. See the following dummy render, compare the left and right connectors. Everything is scaled proportionally to fit the largest element in the table, distorting smaller elements.

Thank you for the dummy render. Now, I understand. Then I suggest you consider `f'<td align="right" >{connector.pincolors[counter].replace("__","")} {html_color_icon(connector.pincolors[counter])}</td>' if connector.pincolors and connector.pincolors[counter] else None,` after all, because the extra space I didn't like is reduced when one of the pin rows have the largest total width. Alternatively, the `align="right"` can be removed, but then the bullets are not perfectly aligned above each other.

Reply:
> I tried, but so far, without any real luck. I tried specifying the `width` of the cell, but it was not respected. Using `fixedsize="true"` also produced weird results (smaller cell, but huge gap _between_ cells).

When reading the [doc](https://graphviz.org/doc/info/shapes.html#html), I believe `width`  is a minimum size that can grow if needed, unless `fixedsize="true"`. Have I misunderstood? Did you get gaps between cells even when you had other cells in the same row without `fixedsize="true"`?

Reply:
This is the result I get when I set `fixedsize="true"` both for the `color` and `pincolor` cells. You see the problem, but playing around with it some more, it also looks a bit like the solution I'm looking for with a black border around a colored cell 😁  perhaps we're on to something... more nested tables?

![test](https://user-images.githubusercontent.com/2447251/89117071-212f7880-d49b-11ea-9c9b-c91a141575ef.png)

I'm using
```python
colorbar = f' bgcolor="{wv_colors.translate_color(connector.color, "HEX")}" width="4" fixedsize="true"></td>'
```
and
```python
                                    f'<td bgcolor="{wv_colors.translate_color(connector.pincolors[counter], "HEX")}" width="4" fixedsize="true"></td>' if connector.pincolors else None,
```